server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/TxState.java

```diff
@@ -0,0 +1,183 @@
+   private final GlobalTransaction globalTransaction;
+   private final TxStatus status;
+   private final List<WriteCommand> modifications;
+
+   TxState(GlobalTransaction globalTransaction) {
+      this(globalTransaction, TxStatus.RUNNING, null);
+   }
+
+   private TxState(GlobalTransaction globalTransaction, TxStatus status, List<WriteCommand> modifications) {
+      this.globalTransaction = globalTransaction;
+      this.status = status;
+      this.modifications = modifications;
+   }
+
+   private static TxStatus valueOf(int i) {
+      return CACHED_VALUE[i];
+   }
+
+   private static List<WriteCommand> copyModifications(Collection<WriteCommand> modifications) {
+      return modifications == null || modifications.isEmpty() ?
+            null :
+            Collections.unmodifiableList(new ArrayList<>(modifications));
+   }
+
+   private static List<WriteCommand> unmarshallModifications(ObjectInput input)
+         throws IOException, ClassNotFoundException {
+      List<WriteCommand> modifications = MarshallUtil.unmarshallCollection(input, ArrayList::new);
+      return modifications == null ? null : Collections.unmodifiableList(modifications);
+   }
+
+   public TxStatus status() {
+      return status;
+   }
+
+   public Address getOriginator() {
+      return globalTransaction.getAddress();
+   }
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o) {
+         return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+         return false;
+      }
+
+      TxState txState = (TxState) o;
+
+      return Objects.equals(status, txState.status) &&
+            Objects.equals(globalTransaction, txState.globalTransaction);
+   }
+
+   @Override
+   public int hashCode() {
+      int result = Objects.hashCode(globalTransaction);
+      result = 31 * result + Objects.hashCode(status);
+      return result;
+   }
+
+   @Override
+   public String toString() {
+      return "TxState{" +
+            "globalTransaction=" + globalTransaction +
+            ", status=" + status +
+            ", modifications=" + modifications +
+            '}';
+   }
+
+   GlobalTransaction getGlobalTransaction() {
+      return globalTransaction;
+   }
+
+   List<WriteCommand> getModifications() {
+      return modifications;
+   }
+
+   TxState prepare(Collection<WriteCommand> modifications) {
+      return status.canAdvance(TxStatus.PREPARED) ?
+            new TxState(globalTransaction, TxStatus.PREPARED, copyModifications(modifications)) :
+            null;
+   }
+
+   TxState commit() {
+      return status.canAdvance(TxStatus.COMMITTED) ?
+            new TxState(globalTransaction, TxStatus.COMMITTED, null) :
+            null;
+   }
+
+   TxState rollback() {
+      return status.canAdvance(TxStatus.ROLLED_BACK) ?
+            new TxState(globalTransaction, TxStatus.ROLLED_BACK, null) :
+            null;
+   }
+
+   public enum TxStatus {
```

Comment:
I don't fancy another enum describing tx state... can't you use `javax.transaction.Status` codes and a little switch in static `canAdvance`?